### ISSUE TITLE
PSR2.Methods.FunctionCallSignature.SpaceBeforeCloseBracket error message being attached to wrong token

### ISSUE THREAD

**Issue by Dacaspex** [squizlabs/PHP_CodeSniffer#3028]
**Describe the bug**
When passing a lambda (arrow) function as a parameter to a function/class which is multi-line, the error phpcs gives is confusing (or simply wrong). Only a single line lambda function is allowed while on the contrary, when specifying solely a lambda (not as a parameter), multi-line is allowed. When the lambda is broken up more, such that the newlines happen at the top-level break points, no error is generated. 

It seems that the generated error is semantically correct, but the message is rather confusing.

**Code sample**
```php
<?php

class Test
{
    public function foo()
    {
        $test = fn() => 42;
        $test = fn() =>
            42;

        $test = myFunc(fn() => 42);
        $test = myFunc(
            fn() => 
                42
        );
        $test = myFunc(fn() =>
            42
        );

        $test = new MyObject(fn() => 42);
        $test = new MyObject(
            fn() => 
                42
        );
        $test = new MyObject(fn() =>
            42
        );
    }
}
```

**Custom ruleset**
None.

**To reproduce**
1. Create a file called `test.php` with the code sample above.
2. Run `phpcs test.php`
3. See error message displayed
```
FILE: /home/<redacted>/file.php
--------------------------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
--------------------------------------------------------------------------------------------------
  3 | ERROR | [ ] Each class must be in a namespace of at least one level (a top-level vendor
    |       |     name)
 16 | ERROR | [x] Expected 0 spaces before closing parenthesis; newline found
 25 | ERROR | [x] Expected 0 spaces before closing parenthesis; newline found
--------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------

Time: 23ms; Memory: 6MB
```

**Expected behavior**
I expected the multi-line lambdas on line 16 and 25 to generate an error, but with a more straightforward message.

**Versions (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - PHP: 7.4.8 (cli) (built: Jul 13 2020 16:45:47) ( NTS )
 - PHPCS: 3.5.5
 - Standard: PSR2, but occurs in other standards as well.

**Additional context**
None.

**Comment by gsherwood** [comment]
There are a couple of things going on here.

The first thing is that the PSR2 standard (and other likes it) see multi-line arrow functions, arrays, and closures as a single continuous argument. They ignore the newlines inside the definition and so count your examples as single-line definitions, even though they clearly span multiple lines. The PSR2 authors wrote a meta document to clarify this: https://www.php-fig.org/psr/psr-2/meta/

So the error message you are getting is telling you that the closing parenthesis of the function call needs to be on the same line that the param ends on. The message looks like it is working correctly. There is actually nothing wrong with the arrow function itself.

But the line that the error message is being generated on is very confusing. It's actually pointing to the function call itself, even though it is talking about the closing parenthesis. This is because it's normally looking for code like this:
```php
$result = foo( $arg );
```
But it also works for newlines, so the target of that error message needs to be adjusted. This is a very easy change, but it will break any custom unit tests people have, so it's best left for 3.6.0.

With the change, you get these errors on your sample code:
```
FILE: temp.php
----------------------------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------------------------------------
  3 | ERROR | [ ] Each class must be in a namespace of at least one level (a top-level vendor name)
 18 | ERROR | [x] Expected 0 spaces before closing parenthesis; newline found
 27 | ERROR | [x] Expected 0 spaces before closing parenthesis; newline found
----------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------------
```
If you agree those errors are now clearer, I'll schedule this change for 3.6.0.

**Comment by Dacaspex** [comment]
That is indeed more clear. Thanks for the quick feedback!

**Comment by gsherwood** [comment]
Thanks for the feedback. I've added this to the 3.6.0 milestone.

**Comment by gsherwood** [comment]
The error message has been changed to report itself on the closing parenthesis now, as described above. Thanks for reporting this.
